nssdb/certificate.go

```diff
@@ -87,6 +87,16 @@ func (db *NSSDB) AddCertificate(ctx context.Context, cert *x509.Certificate, nam
 		return 0, 0, err
 	}
 
+	matches, err := db.findByAttr(ctx, CKO_CERTIFICATE, "CKA_ID", cert.SubjectKeyId)
+	if err != nil {
+		return 0, 0, fmt.Errorf("find cka id conflicts: %w", err)
+	}
+	for _, id := range matches {
+		if err := db.DeleteCertificate(ctx, id); err != nil {
+			return 0, 0, fmt.Errorf("delete conflicting certificate %d: %w", id, err)
+		}
+	}
```

Comment:
It might make sense for our use case, but is this really necessary in NSS? The subjectKeyId only indicates that the key used is the same. I think we should be able to store a certificate with the same key. 

To me, it might make sense to replace them if the certificates have the same label, although, not allowing that, as we do in the KMSs is a valid option too.

Replacing a different certificate with the same subjectKeyId should be a decision of the application using this package, not by the package.

Reply:
I copied `pk12util` where importing with the same subject key id deletes the existing certificate and importing with the same label or nickname leaves the original.